hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/common/model/HoodieSparkRecord.java

```diff
@@ -307,7 +309,12 @@ public Option<Map<String, String>> getMetadata() {
 
   @Override
   public Option<HoodieAvroIndexedRecord> toIndexedRecord(Schema recordSchema, Properties prop) {
-    throw new UnsupportedOperationException();
+    if (data == null) {
+      return Option.empty();
+    }
+    StructType structType = schema == null ? AvroConversionUtils.convertAvroSchemaToStructType(recordSchema) : schema;
```

Comment:
I see `HoodieSparkParquetReader#getRecordIterator` always set up the `schema` as null, does it mean we execute the avro -> sruct type conversion per-row, the cache may help here BTW.

Reply:
https://github.com/apache/hudi/pull/13313/files#diff-21ccee08cebace9de801e104f356bf4333c017d5d5c0cac4e3de63c7861f3c13R180 this is where the cache is added